SignalMessaging/attachments/SignalAttachment.swift

```diff
@@ -776,6 +784,59 @@ public class SignalAttachment: NSObject {
             return 0.5
         }
     }
+    
+    private class func removeImageMetadata(attachment: SignalAttachment) -> SignalAttachment {
+        
+        guard let source = CGImageSourceCreateWithData(attachment.data as CFData, nil) else {
+            let attachment = SignalAttachment(dataSource : DataSourceValue.emptyDataSource(), dataUTI: attachment.dataUTI)
+            attachment.error = .missingData
+            return attachment
+        }
+        
+        guard let type = CGImageSourceGetType(source) else {
+            let attachment = SignalAttachment(dataSource : DataSourceValue.emptyDataSource(), dataUTI: attachment.dataUTI)
+            attachment.error = .invalidFileFormat
+            return attachment
+        }
+        
+        let count = CGImageSourceGetCount(source)
+        let mutableData = NSMutableData()
+        guard let destination = CGImageDestinationCreateWithData(mutableData as CFMutableData, type, count, nil) else {
+            attachment.error = .couldNotRemoveMetadata
+            return attachment
+        }
+        
+        let removeMetadataProperties : [String : AnyObject] =
+        [
+            kCGImagePropertyExifDictionary as String : kCFNull,
+            kCGImagePropertyExifAuxDictionary as String : kCFNull,
+            kCGImagePropertyGPSDictionary as String : kCFNull,
+            kCGImagePropertyTIFFDictionary as String : kCFNull,
+            kCGImagePropertyJFIFDictionary as String : kCFNull,
+            kCGImagePropertyPNGDictionary as String : kCFNull,
+            kCGImagePropertyIPTCDictionary as String : kCFNull,
+            kCGImagePropertyMakerAppleDictionary as String : kCFNull
```

Comment:
Things to test

* Orientation is preserved.
* GIF timing is preserved.

*

Reply:
* Color profiles are preservd.

Reply:
Hi! I just re-tested that the color space and profile, as well as orientation is preserved. 

After stripping the data, these fields are left, for example:

    ([{TIFF}: {
	Orientation = 1;
	ResolutionUnit = 2;
	XResolution = 72;
	YResolution = 72;
    }, PixelWidth: 3264, {Exif}: {
	ColorSpace = 1;
	PixelXDimension = 3264;
	PixelYDimension = 2448;
    }, PixelHeight: 2448, {JFIF}: {
	DensityUnit = 0;
	JFIFVersion =     (
	    1,
	    0,
	    1
	);
	XDensity = 72;
	YDensity = 72;
    }, ProfileName: sRGB IEC61966-2.1, DPIWidth: 72, DPIHeight: 72, ColorModel: RGB, Orientation: 1, Depth: 8])


Timing looks good even with the iOS 11 live photo "GIF" movies that can be sent.